insights-notification-schemas-java/src/test/java/com/redhat/cloud/notifications/ingress/TestSerialization.java

```diff
@@ -132,13 +133,30 @@ void shouldHaveDefaultValuesWhenNotSet() {
+         Action action = Parser.decode(jsonAction);
+         assertTrue(action.getEvents().get(0).getAdditionalProperties().containsKey("report_url"));
+
+         assertThrows(ParsingException.class, () -> Parser.encode(action));
+         assertThrows(ParsingException.class, () -> Parser.validate(action));
 
-         assertDoesNotThrow(() -> Parser.decode(Parser.encode(action)));
+         // Should throw ParsingException but does throw ClassCastException
+         // fixed by: https://github.com/networknt/json-schema-validator/pull/561
+         assertThrows(Exception.class, () -> Parser.validate(jsonAction));
```

Comment:
If it throws `ClassCastException`, why not assert that exact type of exception instead of `Exception`?

Reply:
Updated - the ClassCastException wasn't relevant in this context. It was throw because we were sending payload and context as strings - these are only supported on the decode.